js/index.js

```diff
@@ -1,7 +1,246 @@
+import { API } from './API.js';
+import { userTemplate, itemTemplate } from './template.js';
+
+let currentUser = '';
+let currentUserID = '';
+let users = [];
+let todoList = [];
+let todoFilter = 'all';
+
+const userCreateButton = document.querySelector('.user-create-button');
+const $todoList = document.querySelector('ul.todo-list');
+const $userList = document.querySelector('div#user-list');
+let currentUserButton = $userList.querySelectorAll('button.ripple')[0];
+
+const $newTodo = document.querySelector('input.new-todo');
+
+userCreateButton.addEventListener('click', () => onUserCreateHandler());
+$userList.addEventListener('click', event => {
+  $userList.querySelectorAll('button.ripple').forEach(($user, index) => {
+    if($user.contains(event.target) && !$user.classList.contains('user-create-button')){
+      onUserSelectHandler($user, index);
+      // onUserDeleteHandler($user, index)
+    }
+  });
+})
+
+$todoList.addEventListener('click', event => {
+  $todoList.querySelectorAll('input.toggle').forEach(($item, index) => {
+    if($item.contains(event.target)){
+      toggleTodo($item, index);
+    }
+  })
+  $todoList.querySelectorAll('button.destroy').forEach(($item, index) => {
+    if($item.contains(event.target)){
+      deleteTodo($item, index);
+    }
+  });
+})
+
+$todoList.addEventListener('dblclick', event => {
+  $todoList.querySelectorAll('label.label').forEach(($item, index) => {
+    if($item.contains(event.target)){
+      const itemElement = $item.parentNode.parentNode;
+      itemElement.classList.add('editing');
+    }
+  });
+})
+
+$newTodo.addEventListener('keyup', event => {
+  if(event.key === 'Enter'){
+    addTodo($newTodo.value);
+  }
+})
+
+$todoList.addEventListener('keyup', event => {
+  if(event.key === 'Enter'){
+    $todoList.querySelectorAll('input.edit').forEach(($item, index) => {
+      const itemElement = $item.parentNode;
+      if(itemElement.classList.contains('editing')){
+        modifyTodo($item, index);
+      }
+    })
+  }
+});
+
+window.onload = () => {
+  loadUserList();
+}
+
+const loadUserList = () => {
+  API.fetchUserList()
+    .then(response => response.json())
+    .then(data => {
+      users = data;
+      init();
+    })
+    .catch(err => console.error(err));
+}
+
+const init = () => {
+  renderUsers();
+  currentUserButton = $userList.querySelectorAll('button.ripple')[0];
+  onUserSelectHandler(currentUserButton, 0);
+  render();
+}
+
+const renderUsers = () => {
+  users.forEach(user => addUser(user));
+
+  const currentUserButton = document.querySelector('button.ripple.active');
+  if(currentUserButton){
+    currentUser = currentUserButton.innerText;
+  }
+  else{
+    currentUser = document.querySelector('button.ripple');
+    currentUser.classList.add('active');
+  }
+}
+
 const onUserCreateHandler = () => {
   const userName = prompt("추가하고 싶은 이름을 입력해주세요.");
+  if(userName.length < 2){
+    alert("User의 이름은 최소 2글자 이상이어야 합니다.");
+    return;
+  }
```

Comment:
이 부분은 상수로 분리해도 좋을거 같습니다!
```suggestion
const MINIMUM_USER_NAME_LENGTH = 2;

  if(userName.length < MINIMUM_USER_NAME_LENGTH){
      return alert(`User의 이름은 최소 ${MINIMUM_USER_NAME_LENGTH}글자 이상이어야 합니다.`);
  }
```